plugin/src/main/java/com/denizenscript/denizen/objects/NPCTag.java

```diff
@@ -1307,6 +1399,98 @@ else if (attribute.startsWith("list", 2)) {
                 hologram.addTemporaryLine(text.asString(), duration.getTicksAsInt());
             }
         });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name waypoint_provider
+        // @input ElementTag
+        // @description
+        // Sets the NPC's Waypoints Provider. (default options: linear, wander, guided)
+        // @tags
+        // <NPCTag.waypoint_provider>
+        // -->
+        tagProcessor.registerMechanism("waypoint_provider", false, ElementTag.class, (object, mechanism, input) -> {
+            Waypoints wp = object.getCitizen().getOrAddTrait(Waypoints.class);
+            wp.setWaypointProvider(input.asString());
+        });
+
+        // <--[mechanism]
+        // @object NPCTag
+        // @name wander_delay
+        // @input ElementTag(Integer)
+        // @description
+        // Sets the delay for an NPC's wander Waypoints Provider.  Set to -1 to use Mincraft's default value.
```

Comment:
stray doublespace here

Also meta should say something about the need to set the provider type first